client/tokenized-payment-request/event-handlers.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import {
+	transformStripePaymentMethodForStoreApi,
+	transformStripeShippingAddressForStoreApi,
+} from './transformers/stripe-to-wc';
+import {
+	transformCartDataForDisplayItems,
+	transformCartDataForShippingOptions,
+	transformPrice,
+} from './transformers/wc-to-stripe';
+
+import {
+	getPaymentRequestData,
+	getErrorMessageFromNotice,
+} from './frontend-utils';
+
+import PaymentRequestCartApi from './cart-api';
+
+const cartApi = new PaymentRequestCartApi();
+
+export const shippingAddressChangeHandler = async ( event ) => {
+	try {
+		// Please note that the `event.shippingAddress` might not contain all the fields.
+		// Some fields might not be present (like `line_1` or `line_2`) due to semi-anonymized data.
+		const cartData = await cartApi.updateCustomer(
+			transformStripeShippingAddressForStoreApi( event.shippingAddress )
+		);
+
```

Comment:
```suggestion
				const shippingOptions = transformCartDataForShippingOptions(
					cartData
				);

				// when no shipping options are returned, the API still returns a 200 status code.
				// We need to ensure that shipping options are present - otherwise the PRB dialog won't update correctly.
				if ( shippingOptions.length === 0 ) {
					event.updateWith( {
						// Possible statuses: https://docs.stripe.com/js/appendix/payment_response#payment_response_object-complete
						status: 'invalid_shipping_address',
					} );
					_self.cachedCartData = cartData;

					return;
				}
```
This seems to be borrowed from [here](https://github.com/Automattic/woocommerce-payments/blob/develop/client/tokenized-payment-request/payment-request.js#L256) with the omission of [this little section of code](https://github.com/Automattic/woocommerce-payments/blob/develop/client/tokenized-payment-request/payment-request.js#L266-L280). 

I noticed that when my store did not have any shipping options enabled, the EPM modal initially hung and then produced the following error.

![EPM error](https://cdn-std.droplr.net/files/acc_1104206/MPI01F)
_Uh oh_

Do we maybe need to return this section of code back to account for this scenario? Is there a particular reason why it was initially removed? 🤔

Reply:
>Do we maybe need to return this section of code back to account for this scenario? Is there a particular reason why it was initially removed? 

Yeah, we still need it. I must've overlooked it. With it added in, this error is now handled correctly - [a1cf344](https://github.com/Automattic/woocommerce-payments/pull/9150/commits/a1cf344b38d7732f7e942a1daeaabf5d3d203613)